just_curious_game.ex

```diff
@@ -0,0 +1,325 @@
+    end
+
+    :ok
+  end
+
+  ## PARTICIPANTS ##
+  @spec players(%JustCurious.Game{}) :: [String.t]
+  def players(game), do: smembers("team:#{game.team}:game:#{game}:players") |> Enum.sort()
+
+  @spec spectators(%JustCurious.Game{}) :: [String.t]
+  def spectators(game), do: smembers("team:#{game.team}:game:#{game}:spectators") |> Enum.sort()
+
+  @spec joins(%JustCurious.Game{}) :: [String.t]
+  def joins(game), do: smembers("team:#{game.team}:game:#{game}:joins") |> Enum.sort()
+
+  @spec invites(%JustCurious.Game{}) :: [String.t]
+  def invites(game), do: smembers("team:#{game.team}:game:#{game}:invites") |> Enum.sort()
+
+  @spec actives(%JustCurious.Game{}) :: [String.t]
+  def actives(game), do: joins(game) ++ players(game)
+
+  @spec participants(%JustCurious.Game{}) :: [String.t]
+  def participants(game) do
+    sunion([
+      "team:#{game.team}:game:#{game}:players",
+      "team:#{game.team}:game:#{game}:spectators",
+      "team:#{game.team}:game:#{game}:joins",
+      "team:#{game.team}:game:#{game}:invites" # TODO Should invites be included here?
+    ])
+  end
+
+  ## COUNTS ##
+  @spec active_player_count(%JustCurious.Game{}) :: integer
+  def active_player_count(game), do: actives(game) |> length()
+
+  @spec submission_count(%JustCurious.Game{}) :: integer
+  def submission_count(game), do: "team:#{game.team}:game:#{game}:submissions" |> zcard |> String.to_integer
+
+  @spec nomination_count(%JustCurious.Game{}) :: integer
+  def nomination_count(game), do: "team:#{game.team}:game:#{game}:submissions" |> zcount(1, "+inf") |> String.to_integer
+
+  @spec answer_count(%JustCurious.Game{}) :: integer
+  def answer_count(game) do
+    players(game)
+      |> Enum.map(fn(user) -> hget("team:#{game.team}:game:#{game}:participant:#{user}", "answer_#{game.chainlink}") end)
+      |> Enum.reject(&(&1 == :undefined))
+      |> Enum.count
+  end
+
+  ## TIES / DEADLOCK ##
+  @spec tied?(%JustCurious.Game{}) :: boolean
+  def tied?(game) do
+    top_two = zrevrange("team:#{game.team}:game:#{game}:submissions", 0, -1)
+      |> Enum.take(2)
+      |> Enum.map(&(zscore("team:#{game.team}:game:#{game}:submissions", &1)))
+
+    nomination_count(game) > 0 and Enum.at(top_two, 0) == Enum.at(top_two, 1)
+  end
+
+  @spec break_tie(%JustCurious.Game{}) :: %JustCurious.Question{}
+  def break_tie(game) do
+    submitters       = zrevrange("team:#{game.team}:game:#{game}:submissions", 0, -1)
+    high_score       = zscore("team:#{game.team}:game:#{game}:submissions", List.first(submitters)) |> String.to_integer()
+    submitters
+      |> Enum.map(&({zscore("team:#{game.team}:game:#{game}:submissions", &1) |> String.to_integer(), &1}))
+      |> Enum.sort()
+      |> Enum.reject(fn({score, _submitter}) -> score < high_score end)
+      |> Enum.map(fn({_score, submitter}) -> submitter end)
+      |> roll_the_dice(game)
+  end
+
+  @spec roll_the_dice([String.t], %JustCurious.Game{}) :: :ok
```

Comment:
Is this typespec accurate? I don't see how the last line, `JustCurious.Repo.get(JustCurious.Question, question_id)`, could return just `:ok`

Reply:
Yep, you probably want to typespec the return as a `Game.t`, if it's defined as a proper typespec'ed `@type`.